### PR TITLE
[process] Collect the basics per-process from /proc/$PID/

### PR DESCRIPTION
**Pull request by BryanQuigley**
This includes the limits, the command, status, cpuset,
stack, environment info, and out of memory scores.

Signed-off-by: Bryan Quigley <Bryan.Quigley@Canonical.com>

### CODE REVIEW

File: sos/plugins/process.py [hunk 1]
```diff
@@ -30,5 +30,13 @@ def setup(self):
             "ps auxwwwm",
             "ps alxwww"
         ])
+        self.add_copy_spec([
+            "/proc/*/comm",
+            "/proc/*/limits",
+            "/proc/*/stack",
+            "/proc/*/status",
+            "/proc/*/cpuset",
+            "/proc/*/oom_*"
```
Comment (battlemidget) [inline sos/plugins/process.py]: @bmr-cymru how do you feel about this?

Comment (bmr-cymru) [inline sos/plugins/process.py]: I can see uses for it; the main drawback is afaik there's no support to point regular tools like `ps` at a snapshot of this data. I'd be happy for it to go in as long as we have some benchmarking numbers that confirm it's not going to add significantly to the runtime (the size of the data should not be a problem - even on systems with 1000s of running processes).

**Comment by BryanQuigley** [comment]
With an extreme case (30,000 processes) it does run into some issues (just doing process plugin):
Stock - 47s, file size 892K
With above - 47 minutes, file size at end 6.9M.   Does get up to 300+ megs during compression!
With above --build ,  just 3m41s

I'll do some more tests and see if I can't reduce the total that needs to be compressed.

**Comment by battlemidget** [comment]
@BryanQuigley any updates from your last comment?

**Comment by BryanQuigley** [comment]
I haven't found any good way forward.  Any preference?
Options: 
Just collect the first 1000 processes 
Write a bit of code to go through things that might be empty or the same (oom_, limits) and only record it once, and then say these PIDs also have that.

**Comment by battlemidget** [comment]
@BryanQuigley we should probably move this over to the issues section and continue the discussion there

**Pull request closed without merging**